app/apollo/resolvers/channel.js

```diff
@@ -629,14 +654,26 @@ const channelResolvers = {
 
         // Create output for graphQL plugins
         const subscriptionObjs = _.map(createdSubscriptions, (subscription)=>{
-          return {
-            name: subscription.name,
-            uuid: subscription.uuid,
-          };
+          if (subscription.groups.length > 0) {
+            return {
+              name: subscription.name,
+              uuid: subscription.uuid,
+              groups: subscription.groups
+            };
+          }
+          else {
+            return {
+              name: subscription.name,
+              uuid: subscription.uuid,
+            };
+          }
         });
 
         // Allow graphQL plugins to retrieve more information. addChannelVersion can create versions, and subscriptions. Include details of each created resource in pluginContext.
         context.pluginContext = {channel: { name: newVersionObj.channelName, uuid: newVersionObj.channel_id }, version: {name: newVersionObj.name, uuid: newVersionObj.uuid}, subscriptions: subscriptionObjs };
+        if (newVersionObj.description) context.pluginContext.version.description = newVersionObj.description;
+        if (channel.data_location) context.pluginContext.channel.data_location = channel.data_location;
+        if (channel.tags.length > 0) context.pluginContext.channel.tags = channel.tags;
```

Comment:
- Always including details keeps the code more concise and maintainable 
- Including `data_location` unnecessary/undesirable as it cannot be changed in this mutation
- Using `channel.tags` incorrectly uses the _original_ tags, not new ones.
```suggestion
        context.pluginContext = {channel: { name: newVersionObj.channelName, uuid: newVersionObj.channel_id, tags: tags }, version: {name: newVersionObj.name, uuid: newVersionObj.uuid, description: newVersionObj.description}, subscriptions: subscriptionObjs };
```

Reply:
Writing just 'tags' doesn't work here. It's not specified at any point above. Doing channel.tags finds the 'tags' with given uuid and org_id